testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReader.java

```diff
@@ -651,11 +647,43 @@ public void addColumn(String name, RelDataType type) {
 
     public void addColumn(String name, RelDataType type, boolean isKey) {
       if (isKey) {
-        keyList.add(columnList.size());
+        keyList.add(ImmutableBitSet.of(columnList.size()));
       }
       columnList.add(Pair.of(name, type));
     }
 
+    public void addKey(String... columns) {
```

Comment:
nitpick: what about `addKeys`?

Reply:
The method adds one key (over possibly multiple columns) so using singular noun seems more appropriate.